dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/CompanyOwnershipChecker.kt

```diff
@@ -56,4 +58,44 @@ class CompanyOwnershipChecker(
         val companyId = dataMetaInformationManager.getDataMetaInformationByDataId(dataId).company.companyId
         return isCurrentUserCompanyOwnerForCompany(companyId)
     }
+
+    /**
+     * Method to check whether a given company has at least one company owner
+     * @param companyId the ID of the company
+     * @return a Boolean indicating whether the company has at least one company owner
+     */
+    @Transactional(readOnly = true)
+    fun companyExistsAndHasNoOwner(companyId: String): Boolean {
+        val companyOwners = companyRolesControllerApi.getCompanyRoleAssignments(
+            CompanyRole.CompanyOwner, UUID.fromString(companyId),
+        )
+        return companyOwners.isEmpty()
+    }
+
+    // This function can be made more generic if additional field-specific checks are needed in the future
+    /**
+     * Method to check whether the patch contains only fields that are allowed to be altered by the uploader
+     * @param patch the fields to be patched
+     * @return a Boolean indicating whether the patch complies with the access requirements
+     */
+    @Transactional(readOnly = true)
+    fun onlyPatchesAuthorizedFields(patch: CompanyInformationPatch): Boolean {
+        val unauthorizedFields = mutableListOf<String>()
+
+        if (patch.companyName != null) unauthorizedFields.add("companyName")
+        if (patch.companyAlternativeNames != null) unauthorizedFields.add("companyAlternativeNames")
+        if (patch.companyLegalForm != null) unauthorizedFields.add("companyLegalForm")
+        if (patch.headquarters != null) unauthorizedFields.add("headquarters")
+        if (patch.headquartersPostalCode != null) unauthorizedFields.add("headquartersPostalCode")
+        if (patch.sector != null) unauthorizedFields.add("sector")
+        if (patch.identifiers != null) unauthorizedFields.add("identifiers")
+        if (patch.countryCode != null) unauthorizedFields.add("countryCode")
+        if (patch.isTeaserCompany != null) unauthorizedFields.add("isTeaserCompany")
+        if (patch.parentCompanyLei != null) unauthorizedFields.add("parentCompanyLei")
+
+        if (unauthorizedFields.isNotEmpty()) {
+            throw AccessDeniedException(logMessageBuilder.generateInvalidAlterationExceptionMessage(unauthorizedFields))
+        }
+        return true
```

Comment:
Ich glaube das geht viel leaner. Vorschlag:

![image](https://github.com/d-fine/Dataland/assets/99255296/d4a6f5d6-03e2-4ecf-bd17-62ff492300f3)

Reply:
updated the function based on your suggestion

Reply:
Habe ein weiteres Problem gerade entdeckt:

Der Text von "generateInvalidAlterationExceptionMessage" kommt nicht beim user an, sondern geht im stacktrace verloren:

![image](https://github.com/d-fine/Dataland/assets/99255296/a385b42e-6d66-423e-995e-0cbc3ddf5b0c)

Der stacktrace ist aber auf dataland.com nicht zu sehen.  Ich sehe ihn hier nur, weil das Backend lokal bei mir im dev Profil läuft.

Wenn wir wollen, dass der user diesen Text in seiner response lesen kann, dann müssen wir glaube ich eine custom api exception dafür einführen (ähnlich zur z.B. ResourceNotFoundException), damit Spring das in die response packt.

Reply:
do we even need to explicitly mention what fields the user is not allowed to patch? I just had this as a fancy additional info

Reply:
added the new exception type

Reply:
Lass uns telefonieren dazu

Reply:
Guck ich mir nochmal an

Reply:
Funktioniert 👍 

Bitte noch den return type "Boolean" entfernen und das `return true` in der letzten Zeile entfernen.
Das ist nicht notwendig, weil du ja eine exception schmeißt, wenn unauthorized fields verändert werden.

Reply:
done